ext/opentelemetry-ext-zipkin/src/opentelemetry/ext/zipkin/__init__.py

```diff
@@ -0,0 +1,182 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Zipkin Span Exporter for OpenTelemetry."""
+
+import json
+import logging
+from typing import Optional, Sequence
+
+import requests
+
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.trace import Span, SpanContext, SpanKind
+
+DEFAULT_ENDPOINT = "/api/v2/spans"
+DEFAULT_HOST_NAME = "localhost"
+DEFAULT_PORT = 9411
+DEFAULT_PROTOCOL = "http"
+ZIPKIN_HEADERS = {"Content-Type": "application/json"}
+
+SPAN_KIND_MAP = {
+    SpanKind.INTERNAL: None,
+    SpanKind.SERVER: "SERVER",
+    SpanKind.CLIENT: "CLIENT",
+    SpanKind.PRODUCER: "PRODUCER",
+    SpanKind.CONSUMER: "CONSUMER",
+}
+
+SUCCESS_STATUS_CODE = (200, 202)
+
+logger = logging.getLogger(__name__)
+
+
+class ZipkinSpanExporter(SpanExporter):
+    """Zipkin span exporter for OpenTelemetry.
+
+    Args:
+        service_name: Service that logged an annotation in a trace.Classifier
+            when query for spans.
+        host_name: The host name of the Zipkin server
+        port: The port of the Zipkin server
+        endpoint: The endpoint of the Zipkin server
+        protocol: The protocol used for the request.
+        ipv4: Primary IPv4 address associated with this connection.
+        ipv6: Primary IPv6 address associated with this connection.
+    """
+
+    def __init__(
+        self,
+        service_name: str,
+        host_name: str = DEFAULT_HOST_NAME,
+        port: int = DEFAULT_PORT,
+        endpoint: str = DEFAULT_ENDPOINT,
+        protocol: str = DEFAULT_PROTOCOL,
+        ipv4: Optional[str] = None,
+        ipv6: Optional[str] = None,
+    ):
+        self.service_name = service_name
+        self.host_name = host_name
+        self.port = port
+        self.endpoint = endpoint
+        self.protocol = protocol
+        self.url = "{}://{}:{}{}".format(
+            self.protocol, self.host_name, self.port, self.endpoint
+        )
+        self.ipv4 = ipv4
+        self.ipv6 = ipv6
+
+    def export(self, spans: Sequence[Span]) -> "SpanExportResult":
+        zipkin_spans = self._translate_to_zipkin(spans)
+        result = requests.post(
+            url=self.url, data=json.dumps(zipkin_spans), headers=ZIPKIN_HEADERS
+        )
+
+        if result.status_code not in SUCCESS_STATUS_CODE:
+            logger.error(
+                "Traces cannot be uploaded; status code: %s, message %s",
+                result.status_code,
+                result.text,
+            )
+            # TODO: should we retry here?
```

Comment:
maybe it's good as a configuration variable?

Reply:
good idea, i added a `retry` option to configure the behaviour